### PR TITLE
improve: showing showing server description in dropdown

### LINKED ISSUE #4802: Missing description for `servers:` definition
### Q&A
 - OS: macOS
 - Browser: chrome
 - Version: 68
 - Method of installation: npm
 - Swagger-UI version: 3.18.0
 - Swagger/OpenAPI version: OpenAPI 3.0

### Content & configuration
screen shot 2018-08-08 at 11 56 42

Example Swagger/OpenAPI definition:
```yaml
servers:
- url: https://{customerkey}-{projectkey}.digitalcx.com
  description: The production API server
#  variables:
#    customerkey:
#      default: something
#      description: This is your customer key (provided when the app was setup)
#    projectkey:
#      default: something
#      description: This is your project key (provided when the app was setup)
```

Swagger-UI configuration options:
I am using `swagger-ui-express` but nothing else on the SwaggerUI() function configuration

### Describe the bug you're encountering
I expect the description for the server to be there as defined in OpenApi 3, but instead I don't see anything.
Same is happening when I add the `customerkey` and the `projectkey` fields

### To reproduce...

1. Copy te yaml provided here
2. Open the swagger ui and preview the changes
3. See error

### Expected behavior
I would expect the provided description to be showing up underneath the resource, or at least somewhere in the UI.

### Screenshots
screen shot 2018-08-08 at 11 56 42

### Additional context or thoughts
Could this be a feature non implemented?
Could this be related with `swagger-ui-express`?

Thank you very much

### PR DESCRIPTION
**Pull request by dersvenhesse**
### Description
This PR adds OpenAPI3's server description to the dropdown when the description ist set. The URLs get a bit more context and information due to the additional output.

I figured it'd be best to append the description to the URL instead of prepending it, so all `https://...`'s are still aligned. Parentheses are used for separation.

### Motivation and Context
Fixes #4802.

### How Has This Been Tested?
Since the code is pretty simple I just tested the ui manually. The options value remains unchanged, so there should be no side effects.

### Screenshot:
bildschirmfoto 2018-11-08 um 23 27 43

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dersvenhesse** [comment]
@shockey I‘m sorry, but I‘d be happy to hear from you. ;)
Is this feature generally mergeable? Is there anything missing? Thanks a lot!

**Comment by shockey** [comment]
Hey @dersvenhesse! No worries, sometimes a nudge is well-placed 😄 

I like the feature, but the code is a bit less readable than it could be. I'll make an inline comment with details...

File: src/core/plugins/oas3/components/servers.jsx [hunk 1]
```diff
@@ -110,7 +110,7 @@ export default class Servers extends React.Component {
               <option
                 value={ server.get("url") }
                 key={ server.get("url") }>
-                { server.get("url") }
+                { server.get("url") } { server.get("description") ? "("+server.get("description")+")" : "" }
```
Comment (shockey) [inline src/core/plugins/oas3/components/servers.jsx]: This would be better as a [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) 😄

**Comment by shockey** [comment]
Also - I am slightly concerned that `()` is similar to `{}`, which could be visually confusing.

@webron, any 💡 bright ideas here?

**Comment by webron** [comment]
@shockey making the description font bold (or the url itself) should make it look different enough.

**Comment by dersvenhesse** [comment]
@webron, unfortunately bold formatting in `option`s doesn't work. In my knowledge swagger-ui uses default dropdowns without js and you can't put other tags inside such. 

Here are some other ideas:
* http://petstore-depr.swagger.io/v1 – Deprecated server
* http://petstore-depr.swagger.io/v1 : Deprecated server
* http://petstore-depr.swagger.io/v1    Deprecated server

File: src/core/plugins/oas3/components/servers.jsx (hunk 1 above)
Comment (dersvenhesse) [reply src/core/plugins/oas3/components/servers.jsx]: Thanks, I will look into that and change it. 👍
I looked in other files beforehand and found similar code style, like [that](https://github.com/swagger-api/swagger-ui/blob/master/src/core/plugins/oas3/components/request-body.jsx#L93).

**Comment by dersvenhesse** [comment]
@shockey and @webron: What's the current status? Should I do something else to improve this?

**Comment by shockey** [comment]
Merged (with slight changes to use a template string and the ` - ` format).

Thanks @dersvenhesse! 🤜 🤛
